runtime/interpreter/statictype.go

```diff
@@ -255,6 +256,21 @@ func NewVariableSizedStaticType(
 	}
 }
 
+func (t VariableSizedStaticType) IsComposite() bool {
+	return false
+}
+
+func (t VariableSizedStaticType) Copy() atree.TypeInfo {
+	// TODO: is creating a new VariableSizedStaticType necessary?
+	// atree needs immutable TypeInfo or a copy of mutable TypeInfo because TypeInfo can be shared.
+	return t
```

Comment:
This might only do a shallow copy (of the element type `t.Type`) since certain types (e.g: RestrictedStaticType) have pointer receivers.

Reply:
Does the `TypeInfo` returned by this method get mutated by atree?

Reply:
> Does the TypeInfo returned by this method get mutated by atree?

No, atree doesn't mutate `TypeInfo`.  

`Copy` is used to prevent mutation of shared `TypeInfo` in case it is mutated by Cadence.  However, if `TypeInfo` is immutable, there is no reason to create a new object.    

@SupunS Can TypeInfo be mutated by Cadence?

Reply:
Ah ok, thanks for the explanation.
TypeInfo is not mutated by cadence. So I guess then this should be sufficient 👍